dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/data/DefaultAnalyticsService.java

```diff
@@ -736,7 +737,7 @@ private void addReportingRates( DataQueryParams params, Grid grid, ReportingRate
                 List<String> dataRow = Lists.newArrayList( entry.getKey().split( DIMENSION_SEP ) );
 
                 Double target = entry.getValue();
-                Double actual = dataMap.get( entry.getKey() );
+                Double actual = ObjectUtils.firstNonNull( dataMap.get( entry.getKey() ), 0d );
 
                 if ( target != null && ( actual != null || metric == EXPECTED_REPORTS ) )
```

Comment:
Minor: I think `actual` cannot be null at this point so lets just remove that null check for code clarity.